src/misc/programs.cpp

```diff
@@ -97,12 +97,10 @@ static Bitu PROGRAMS_Handler(void) {
 	                         256 + static_cast<uint16_t>(exec_block_size));
 	HostPt writer=(HostPt)&index;
 	for (;size>0;size--) *writer++=mem_readb(reader++);
-	Program * new_program;
 	if (index >= internal_progs.size()) E_Exit("something is messing with the memory");
-	PROGRAMS_Main * handler = internal_progs[index];
-	(*handler)(&new_program);
+	PROGRAMS_Creator creator = internal_progs[index];
```

Comment:
`... = internal_progs.at(index);`
Would let us drop the `if` and get the bounds checking.

In theory, we could have these "something is messing with memory so hard-exit" checks around every `vector/array` index calls, so I'd rather switch to `at(index)` and have it handled by the program catch routine (in which case, we'll get a backtrace).

Reply:
Is it possible to use a reference instead of a copy?
(Maybe even a a `const &`?)